1_demo_hello.bal

```diff
@@ -6,10 +6,21 @@
 
 import ballerina/http;
 
+// Services, endpoint, resources are built into the language
+// This one is HTTP (other options include WebSockets, Protobuf, gRPC, etc)
+// We bind it to port 9090
 service<http:Service> hello bind {port:9090} {
+
+  // The service exposes one resource (hi)
+  // It gets the endpoint that called it - so we can pass response back
+  // and the request struct to extract payload, etc.
    hi (endpoint caller, http:Request request) {
-       http:Response res;
-       res.setPayload("Hello World!\n");
-       _ = caller->respond(res);
+        // Create the Response object
+        http:Response res;
+        // Put the data into it
+        res.setPayload("Hello World!\n");
+        // Send it back. -> means remote call (. means local)
```

Comment:
It is better to add backticks around `->` and `.` so they are highlighted properly. Also better to update the last sentence as (`.` means local **call**)